### ISSUE TITLE
Don't know what to do in Unity

### ISSUE THREAD

**Issue by martincuinet** [kyy13/mc-mesher#20]
Hi !
I followed the unity quick start

    Download the latest release from the releases page.
    Drop the DLL into the your Assets folder
    Drop the mc_mesher.cs wrapper file into your Assets folder.

What's next ? How do I generate the cave ?

Thx

**Comment by kyy13** [comment]
Hi @martincuinet,
I added a minimal example behavior to the repo at `example/unity/McMeshBehaviour.cs`
The instructions are commented at the top of the file.
Unity doesn't have a 3D noise function by default, so it requires the Unity.Mathematics package to generate the noise field.

**Comment by martincuinet** [comment]
Thanks for the response !
I think I did everything right. No error when playing, Mathematics package installed, the cs script on a gameObject with a mesh filter component.
But nothing happens
![Capture](https://github.com/user-attachments/assets/85037fb4-ac1d-4932-bc6c-6c4dbdb48e2a)

Also when I try to attach mc_mesher.cs script to a gameObject, I get 
![image](https://github.com/user-attachments/assets/0d8c5be4-c4ed-4ee8-98da-a6d6a4efa41c)

Any idea ?

**Comment by kyy13** [comment]
@martincuinet That's normal for `mc_mesher.cs`. It should be in your project, but it should not be attached to a GameObject.

Please verify both `mc_mesher.cs` and `libmcmesher.dll` are both somewhere in your Assets folder.

I found an issue with the 1.2.0.win64 release. The `mc_mesher.cs` file in the `wrappers/unity` folder is from a different release version. Please use `mc_mesher.cs` from the main branch, or the `mc_mesher.cs` file from the 1.2.0 source code release instead.

I will patch the 1.2.0 release and notify. Thank you for bringing this to my attention.

**Comment by kyy13** [comment]
The 1.2.0 release is now patched. If you haven't already downloaded the `mc_mesher.cs` file from the repo, then it's also available in the release.

**Comment by martincuinet** [comment]
Hi @kyy13 
thanks for the response and update
Unfortunately, after downloading the mc_mesher.cs from the release, it is still not doing much.
What should happen ? How can I launch the mesh generation ?
![image](https://github.com/user-attachments/assets/ff765421-c49b-4419-9c2b-8f3824313b38)
I even tried to rename the mc_mesher.cs to MCmesher.cs as I see it was its name in the previous release, but no luck.

I'm so sorry to annoy you with this, but I'm not sure what to do for make it work.
Can you explain ?

Cheers

**Comment by kyy13** [comment]
Hi @martincuinet, no problem. Just to verify, is your platform 64-bit windows? I'll see if I can recreate the issue in 2020.3.8f1.

**Comment by martincuinet** [comment]
Yes, 64-bit windows
Also I use HDRP render pripeline in my scene

**Comment by kyy13** [comment]
@martincuinet Oops, you will also need to add and configure a MeshRenderer component to see the mesh. It is probably easiest to attach the script to a 3DObject->Cube, which already has a MeshFilter and MeshRenderer setup.

**Comment by martincuinet** [comment]
Ah ! Great !
Yes, the mesh renderer does the trick,... obviously ^^
Thanks for your patience
how do you manage to modify the cave aspect ? ike different noise seed, offsets,... ?

**Comment by kyy13** [comment]
@martincuinet Yeah, noise seed, frequency, scale, offsets, etc. The data set is a 3D field of floats, so it can be anything. Additionally, there's a tolerance that you pass to the library that determines the threshold where you cross the volume's surface, so that can be used to control how much of the mesh is solid vs empty.

If the data point in the field is above the tolerance, then it is inside the surface, and if it is below the tolerance, then it is outside the surface. I believe the example is set to 0.3, and the random function produces values between ~[0,1], so roughly 70% of the volume should be solid in the example.

For the library image, I used a triplanar shader to texture the mesh.
I updated the example to include instructions to add the MeshRenderer. Thanks for helping me improve the library!
